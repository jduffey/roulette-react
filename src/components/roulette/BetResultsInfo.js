import { getWheelNumberColor } from "../../common/getWheelNumberColor";

const CLASS_NAME = "BetResultsInfo-component";
export function BetResultsInfo(props) {
    const previousRoundHasBetResults = props.previousRoundResults !== null;

    let startingBalanceText = "-";
    let finalBalanceText = "-";
    let netChangeInBalanceText = "-";
    let totalBetAmountText = "-";
    let totalWinningsText = "-";
    let totalBetsReturnedText = "-";
    if (previousRoundHasBetResults) {
        startingBalanceText = `$ ${props.previousRoundResults.startingBalance}`;
        finalBalanceText = `$ ${props.previousRoundResults.finalBalance}`;
        netChangeInBalanceText = `$ ${props.previousRoundResults.finalBalance - props.previousRoundResults.startingBalance}`;

        const sumBetAmounts = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.betAmount, 0);
        const sumWinnings = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.winningsOnBet, 0);
        const sumBetsReturned = Object.values(props.previousRoundResults.resultsOfBets).reduce((acc, bet) => acc + bet.betReturned, 0);

        totalBetAmountText = `$ ${sumBetAmounts}`;
        totalWinningsText = `$ ${sumWinnings}`;
        totalBetsReturnedText = `$ ${sumBetsReturned}`;
    }

    const previousWheelNumberDiv = () => {
        const backgroundColor = previousRoundHasBetResults ?
            getWheelNumberColor(props.previousRoundResults.winningWheelNumber)
            : "#dfdfdf";

        const wheelNumberText = previousRoundHasBetResults ?
            props.previousRoundResults.winningWheelNumber :
            "??";

        return (
            <div
                className="bet-info-table-title-spin-result"
                style={{ backgroundColor }}
            >
                {wheelNumberText}
            </div>
        )
    }

    const betResultsRows = () => Object.keys(props.previousRoundResults.resultsOfBets).map((betName) => {
        const betAmountOnBet = props.previousRoundResults.resultsOfBets[betName].betAmount;
        const winningsOnBet = props.previousRoundResults.resultsOfBets[betName].winningsOnBet;
        const betReturnedAmount = props.previousRoundResults.resultsOfBets[betName].betReturned;
        return (
            <tr key={betName}>
                <td>{betName}</td>
                <td className="bet-results-info-table-bet-amount">{`$ ${betAmountOnBet}`}</td>
                <td>{`$ ${winningsOnBet}`}</td>
                <td>{`$ ${betReturnedAmount}`}</td>
            </tr>
        );
    });

    return (
        <div
            className={CLASS_NAME}
        >
            <div
                className="bet-info-table-title-container"
            >
                <div
                    className="bet-info-table-title-text"
                >
                    PREVIOUS ROUND RESULTS
                </div>

                {previousWheelNumberDiv()}

            </div>
            <table className="bet-results-info-table">
                <tbody>
                    <tr style={{
                        textAlign: "center",
                    }}>
                        <th>Bet Name</th>
                        <th>Bet Amount</th>
                        <th>Winnings</th>
                        <th>Bet Returned</th>
                    </tr>

                    {previousRoundHasBetResults ? betResultsRows() : null}

                    <tr style={{ height: "10px" }} />
                    <tr>
                        <td>TOTALS</td>
                        <td>{totalBetAmountText}</td>
                        <td>{totalWinningsText}</td>
                        <td>{totalBetsReturnedText}</td>
                    </tr>
                    <tr style={{ height: "10px" }} />
                    <tr className="balance-change-value">
                        <td>
                            Starting
                        </td>
                        <td colSpan={3}>
                            {startingBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Net
                        </td>
                        <td colSpan={3}>
                            {netChangeInBalanceText}
                        </td>
                    </tr>
                    <tr className="balance-change-value">
                        <td>
                            Final
                        </td>
                        <td colSpan={3}>
                            {finalBalanceText}
                        </td>
                    </tr>
                </tbody>
            </table>
        </div >
    );
}
