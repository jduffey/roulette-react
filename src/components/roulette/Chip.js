import { CHIP_COLORS } from "../../common/standardColors";

const CHIP_AMOUNTS = {
    CHIP_1: 1,
    CHIP_5: 5,
    CHIP_25: 25,
    CHIP_100: 100,
    CHIP_500: 500,
    CHIP_1000: 1000,
};

const getChipColorStyles = (chipAmt) => {
    switch (true) {
        case (chipAmt >= CHIP_AMOUNTS.CHIP_1 && chipAmt < CHIP_AMOUNTS.CHIP_5):
            return Object.assign({}, CHIP_COLORS.CHIP_1);
        case (chipAmt >= CHIP_AMOUNTS.CHIP_5 && chipAmt < CHIP_AMOUNTS.CHIP_25):
            return Object.assign({}, CHIP_COLORS.CHIP_5);
        case (chipAmt >= CHIP_AMOUNTS.CHIP_25 && chipAmt < CHIP_AMOUNTS.CHIP_100):
            return Object.assign({}, CHIP_COLORS.CHIP_25);
        case (chipAmt >= CHIP_AMOUNTS.CHIP_100 && chipAmt < CHIP_AMOUNTS.CHIP_500):
            return Object.assign({}, CHIP_COLORS.CHIP_100);
        case (chipAmt >= CHIP_AMOUNTS.CHIP_500 && chipAmt < CHIP_AMOUNTS.CHIP_1000):
            return Object.assign({}, CHIP_COLORS.CHIP_500);
        case (chipAmt >= CHIP_AMOUNTS.CHIP_1000):
            return Object.assign({}, CHIP_COLORS.CHIP_1000);
        default:
            return { display: "none" };
    }
}

const CLASS_NAME = "Chip-component";
function Chip(props) {
    const styles = getChipColorStyles(props.chipAmount);
    if (props.isSelected) {
        styles.boxShadow = "0px 16px 10px 1px rgba(0, 0, 0, 0.6)";
        styles.marginTop = "12px";
    }

    return (
        <div
            className={`${CLASS_NAME} ${props.auxiliaryClassName}`}
            id={props.id}
            key={props.chipAmount}
            onClick={() => props.onClick(props.chipAmount)}
            style={styles}
        >
            {`$${props.chipAmount}`}
        </div>
    );
}

export { Chip, CHIP_AMOUNTS }
