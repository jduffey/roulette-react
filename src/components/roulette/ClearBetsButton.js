const CLASS_NAME = "ClearBetsButton-component";

export function ClearBetsButton(props) {
    const hasBets = props.pendingBets && props.pendingBets.length > 0;
    const isDisabled = props.wheelIsSpinning || !hasBets;

    return (
        <div className={CLASS_NAME}>
            <button
                className="clear-bets-button"
                onClick={props.onClick}
                disabled={isDisabled}
                style={{
                    backgroundColor: isDisabled ? "#cccccc" : "#ff4444",
                    color: "white",
                    border: "none",
                    borderRadius: "8px",
                    padding: "12px 24px",
                    fontSize: "16px",
                    fontWeight: "bold",
                    cursor: isDisabled ? "not-allowed" : "pointer",
                    opacity: isDisabled ? 0.6 : 1,
                }}
            >
                Clear All Bets
            </button>
        </div>
    );
} 