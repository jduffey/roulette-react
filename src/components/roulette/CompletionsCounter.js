import { useEffect, useState } from 'react';

import {
    getPlayerNumberCompletionSetsCounter,
} from "../../common/blockchainWrapper";

const formattedChainNumber = (chainNumber, decimals) => {
    return chainNumber
        ? parseFloat(chainNumber)
            .toLocaleString(undefined, { minimumFractionDigits: decimals, maximumFractionDigits: decimals })
        : "Loading...";
}

const CLASS_NAME = "CompletionsCounter-component";
export function CompletionsCounter(props) {
    const [completionsCount, setCompletionsCount] = useState(undefined);

    useEffect(() => {
        setTimeout(async () => {
            const count = await getPlayerNumberCompletionSetsCounter(props.playerAddress);
            setCompletionsCount(count);
        }, 1000);
    }, [completionsCount]);

    return (
        <div
            className={CLASS_NAME}
        >
            {formattedChainNumber(completionsCount, 0)}
        </div >
    )
}
