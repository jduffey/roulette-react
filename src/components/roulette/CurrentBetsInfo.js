const betRow = (betOption, betAmount) => {
    return (
        <tr key={betOption}>
            <td>
                {betOption}
            </td>
            <td className="current-bets-info-table-bet-amount">
                {`$ ${betAmount}`}
            </td>
        </tr>
    );
}

const CLASS_NAME = "CurrentBetsInfo-component";
export function CurrentBetsInfo(props) {
    return (
        <div
            className={CLASS_NAME}
        >
            <div
                className="current-bets-info-table-title-text"
            >
                CURRENT BETS
            </div>
            <table className="current-bets-info-table">
                <tbody>
                    <tr style={{
                        textAlign: "center",
                    }}>
                        <th>Bet Name</th>
                        <th>Bet Amount</th>
                    </tr>
                    {Object.keys(props.betsOnBoard).map((betOption) => betRow(betOption, props.betsOnBoard[betOption]))}
                </tbody>
            </table>
        </div>
    );
}
