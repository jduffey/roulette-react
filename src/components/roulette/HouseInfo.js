import { useEffect, useState } from 'react';

import {
    getTokenBalance,
    getTotalSpins,
    getTotalAmountWagered,
    HOUSE_ADDRESS,
    getJackpotBalance,
} from "../../common/blockchainWrapper";

const formattedChainNumber = (chainNumber, decimals) => {
    return chainNumber
        ? parseFloat(chainNumber)
            .toLocaleString(undefined, { minimumFractionDigits: decimals, maximumFractionDigits: decimals })
        : "Loading...";
}

const CLASS_NAME = "HouseInfo-component";
export function HouseInfo() {
    const [houseBalance, setHouseBalance] = useState(undefined);
    const [totalSpins, setTotalSpins] = useState(undefined);
    const [totalAmountWagered, setTotalAmountWagered] = useState(undefined);
    const [jackpotBalance, setJackpotBalance] = useState(undefined);

    useEffect(() => {
        setTimeout(async () => {
            const houseBal = await getTokenBalance(HOUSE_ADDRESS);
            setHouseBalance(houseBal);

            const spins = await getTotalSpins();
            setTotalSpins(spins);

            const totalAmountWagered = await getTotalAmountWagered();
            setTotalAmountWagered(totalAmountWagered);

            const jackpotBal = await getJackpotBalance();
            setJackpotBalance(jackpotBal);
        }, 1000);

    }, [totalSpins]);

    return (
        <div
            className={CLASS_NAME}
        >
            <div>
                House Balance
                < br />
                {formattedChainNumber(houseBalance, 2)}
            </div>
            <div>
                Jackpot Balance
                < br />
                {formattedChainNumber(jackpotBalance, 2)}
            </div>
            <div>
                Total Spins
                < br />
                {formattedChainNumber(totalSpins, 0)}
            </div>
            <div>
                Total Amount Wagered
                < br />
                {formattedChainNumber(totalAmountWagered, 2)}
            </div>
        </div >
    )
}
