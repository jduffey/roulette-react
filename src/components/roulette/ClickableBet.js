import { BET_NAMES } from "../../common/betNames";
import { STANDARD_COLORS } from "../../common/standardColors";

import { Chip } from "./Chip";

const BETTING_SQUARE_OUTLINE_SIZE = 1;

const CLASS_NAME = "ClickableBet-component";
function ClickableBet(props) {
    return (
        <div
            id={props.id}
            className={CLASS_NAME}
            onClick={() => props.onClick()}
            role="button"
            style={{
                left: props.styleData.left,
                top: props.styleData.top,
                width: props.styleData.width - BETTING_SQUARE_OUTLINE_SIZE,
                height: props.styleData.height - BETTING_SQUARE_OUTLINE_SIZE,
            }}
        >
            <div className="betting-square-contents">
                <div
                    className={`betting-square-label ${props.textLabelClassNamePrefix}-square-label`}
                    style={{
                        backgroundColor: props.styleData.labelBackgroundColor,
                    }}
                >
                    {props.displayText}
                </div>
                <Chip
                    auxiliaryClassName="betting-square-chip"
                    key={props.chipAmount}
                    onClick={props.onClick}
                    chipAmount={props.betAmount}
                />
            </div>
        </div>
    );
}

const DISPLAY_PARAMS = Object.entries({
    // StraightUp 1st Column
    [BET_NAMES.STRAIGHT_UP_1]: [80, 240, 80, 120, STANDARD_COLORS.FELT_RED, "1", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_4]: [160, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "4", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_7]: [240, 240, 80, 120, STANDARD_COLORS.FELT_RED, "7", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_10]: [320, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "10", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_13]: [400, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "13", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_16]: [480, 240, 80, 120, STANDARD_COLORS.FELT_RED, "16", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_19]: [560, 240, 80, 120, STANDARD_COLORS.FELT_RED, "19", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_22]: [640, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "22", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_25]: [720, 240, 80, 120, STANDARD_COLORS.FELT_RED, "25", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_28]: [800, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "28", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_31]: [880, 240, 80, 120, STANDARD_COLORS.FELT_BLACK, "31", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_34]: [960, 240, 80, 120, STANDARD_COLORS.FELT_RED, "34", "straightUp1to36"],

    // StraightUp 2nd Column
    [BET_NAMES.STRAIGHT_UP_2]: [80, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "2", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_5]: [160, 120, 80, 120, STANDARD_COLORS.FELT_RED, "5", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_8]: [240, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "8", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_11]: [320, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "11", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_14]: [400, 120, 80, 120, STANDARD_COLORS.FELT_RED, "14", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_17]: [480, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "17", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_20]: [560, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "20", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_23]: [640, 120, 80, 120, STANDARD_COLORS.FELT_RED, "23", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_26]: [720, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "26", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_29]: [800, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "29", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_32]: [880, 120, 80, 120, STANDARD_COLORS.FELT_RED, "32", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_35]: [960, 120, 80, 120, STANDARD_COLORS.FELT_BLACK, "35", "straightUp1to36"],

    // StraightUp 3rd Column
    [BET_NAMES.STRAIGHT_UP_3]: [80, 0, 80, 120, STANDARD_COLORS.FELT_RED, "3", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_6]: [160, 0, 80, 120, STANDARD_COLORS.FELT_BLACK, "6", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_9]: [240, 0, 80, 120, STANDARD_COLORS.FELT_RED, "9", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_12]: [320, 0, 80, 120, STANDARD_COLORS.FELT_RED, "12", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_15]: [400, 0, 80, 120, STANDARD_COLORS.FELT_BLACK, "15", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_18]: [480, 0, 80, 120, STANDARD_COLORS.FELT_RED, "18", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_21]: [560, 0, 80, 120, STANDARD_COLORS.FELT_RED, "21", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_24]: [640, 0, 80, 120, STANDARD_COLORS.FELT_BLACK, "24", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_27]: [720, 0, 80, 120, STANDARD_COLORS.FELT_RED, "27", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_30]: [800, 0, 80, 120, STANDARD_COLORS.FELT_RED, "30", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_33]: [880, 0, 80, 120, STANDARD_COLORS.FELT_BLACK, "33", "straightUp1to36"],
    [BET_NAMES.STRAIGHT_UP_36]: [960, 0, 80, 120, STANDARD_COLORS.FELT_RED, "36", "straightUp1to36"],

    // Zeroes
    [BET_NAMES.STRAIGHT_UP_0]: [0, 180, 80, 180, STANDARD_COLORS.FELT_GREEN, "0", "zero"],
    [BET_NAMES.STRAIGHT_UP_00]: [0, 0, 80, 180, STANDARD_COLORS.FELT_GREEN, "00", "zero"],

    // Dozens
    [BET_NAMES.FIRST_DOZEN]: [80, 360, 320, 80, STANDARD_COLORS.INHERIT, "1st 12", "dozens"],
    [BET_NAMES.SECOND_DOZEN]: [400, 360, 320, 80, STANDARD_COLORS.INHERIT, "2nd 12", "dozens"],
    [BET_NAMES.THIRD_DOZEN]: [720, 360, 320, 80, STANDARD_COLORS.INHERIT, "3rd 12", "dozens"],

    // Columns
    [BET_NAMES.FIRST_COLUMN]: [1040, 240, 80, 120, STANDARD_COLORS.INHERIT, "2 to 1", "column"],
    [BET_NAMES.SECOND_COLUMN]: [1040, 120, 80, 120, STANDARD_COLORS.INHERIT, "2 to 1", "column"],
    [BET_NAMES.THIRD_COLUMN]: [1040, 0, 80, 120, STANDARD_COLORS.INHERIT, "2 to 1", "column"],

    // Halves
    [BET_NAMES.FIRST_18]: [80, 440, 160, 80, STANDARD_COLORS.INHERIT, "1 to 18", "halves"],
    [BET_NAMES.EVEN]: [240, 440, 160, 80, STANDARD_COLORS.INHERIT, "Even", "halves"],
    [BET_NAMES.RED]: [400, 440, 160, 80, STANDARD_COLORS.FELT_RED, "Red", "halves-colors"],
    [BET_NAMES.BLACK]: [560, 440, 160, 80, STANDARD_COLORS.FELT_BLACK, "Black", "halves-colors"],
    [BET_NAMES.ODD]: [720, 440, 160, 80, STANDARD_COLORS.INHERIT, "Odd", "halves"],
    [BET_NAMES.SECOND_18]: [880, 440, 160, 80, STANDARD_COLORS.INHERIT, "19 to 36", "halves"],
}).reduce((acc, [betName, params]) => {
    acc[betName] = {
        styleData: {
            left: params[0],
            top: params[1],
            width: params[2],
            height: params[3],
            labelBackgroundColor: params[4],
        },
        displayText: params[5],
        textLabelClassNamePrefix: params[6],
    };
    return acc;
}, {});

export {
    ClickableBet,
    DISPLAY_PARAMS,
};
