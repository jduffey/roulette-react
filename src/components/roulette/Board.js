import React from 'react';

import { ClickableBet, DISPLAY_PARAMS } from './ClickableBet';

const CLASS_NAME = "Board-component";
export function Board(props) {
    return (
        <div
            className={CLASS_NAME}
        >
            {Object.entries(DISPLAY_PARAMS).map(([betOptionName, params]) => {
                return (
                    <ClickableBet
                        textLabelClassNamePrefix={params.textLabelClassNamePrefix}
                        key={betOptionName}
                        onClick={() => props.onClick(betOptionName)}
                        displayText={params.displayText}
                        betAmount={props.pendingBets.filter(pendingBet => pendingBet.betName === betOptionName).reduce((acc, pendingBet) => acc + pendingBet.betAmount, 0)}
                        styleData={params.styleData}
                    />
                );
            })}
        </div>
    );
}
