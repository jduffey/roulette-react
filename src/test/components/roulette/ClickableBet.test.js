import renderer from 'react-test-renderer';

import { ClickableBet } from '../../../components/roulette/ClickableBet';

describe('ClickableBet', () => {
    describe('renders when', () => {
        it.each([
            ["chip is visible", 1],
            ["chip is not visible", 0],
        ])('renders when %s', (_description, betAmount) => {
            const props = {
                betName: "BET_NAME",
                onClick: () => "ON_CLICK",
                styleData: {
                    left: 10,
                    top: 20,
                    width: 30,
                    height: 40,
                    backgroundColor: "BACKGROUND_COLOR",
                    labelBackgroundColor: "LABEL_BACKGROUND_COLOR",
                },
                textLabelClassNamePrefix: "TEXT_LABEL_CLASSNAME_PREFIX",
                displayText: "DISPLAY_TEXT",
                betAmount,
            };

            const sut = ClickableBet(props);

            const actual = renderer.create(sut);

            expect(actual).toMatchSnapshot();
        });
    });
});
