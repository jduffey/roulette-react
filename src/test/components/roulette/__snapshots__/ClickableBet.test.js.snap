// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableBet renders when renders when chip is not visible 1`] = `
<div
  className="ClickableBet-component"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "height": 39,
      "left": 10,
      "top": 20,
      "width": 29,
    }
  }
>
  <div
    className="betting-square-contents"
  >
    <div
      className="betting-square-label TEXT_LABEL_CLASSNAME_PREFIX-square-label"
      style={
        Object {
          "backgroundColor": "LABEL_BACKGROUND_COLOR",
        }
      }
    >
      DISPLAY_TEXT
    </div>
    <div
      className="Chip-component betting-square-chip"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
    >
      $0
    </div>
  </div>
</div>
`;

exports[`ClickableBet renders when renders when chip is visible 1`] = `
<div
  className="ClickableBet-component"
  onClick={[Function]}
  role="button"
  style={
    Object {
      "height": 39,
      "left": 10,
      "top": 20,
      "width": 29,
    }
  }
>
  <div
    className="betting-square-contents"
  >
    <div
      className="betting-square-label TEXT_LABEL_CLASSNAME_PREFIX-square-label"
      style={
        Object {
          "backgroundColor": "LABEL_BACKGROUND_COLOR",
        }
      }
    >
      DISPLAY_TEXT
    </div>
    <div
      className="Chip-component betting-square-chip"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#DFDFDF",
          "borderColor": "#000000",
          "color": "#000000",
        }
      }
    >
      $1
    </div>
  </div>
</div>
`;
